crates/re_arrow_store/src/store_gc.rs

```diff
@@ -221,32 +231,39 @@ impl DataStore {
             self.metadata_registry.heap_size_bytes -= metadata_dropped_size_bytes;
             num_bytes_to_drop -= metadata_dropped_size_bytes as f64;
 
-            row_ids.push(*row_id);
+            deleted.row_ids.insert(*row_id);
 
             // find all tables that could possibly contain this `RowId`
-            let temporal_tables = self.tables.iter_mut().filter_map(|((timeline, _), table)| {
-                timepoint.get(timeline).map(|time| (*time, table))
-            });
 
-            for (time, table) in temporal_tables {
-                num_bytes_to_drop -= table.try_drop_row(*row_id, time.as_i64()) as f64;
+            for ((timeline, ent_path_hash), table) in &mut self.tables {
+                if let Some(time) = timepoint.get(timeline) {
+                    num_bytes_to_drop -= table.try_drop_row(*row_id, time.as_i64()) as f64;
```

Comment:
Fixing this is somewhat annoying.

The point where we currently have information about whether or not the component is cleared is down in:
https://github.com/rerun-io/rerun/blob/main/crates/re_arrow_store/src/store_gc.rs#L509-L511

Based on whether or not the row_index of that component is Some